flow/core/kernel/vehicle/traci.py

```diff
@@ -421,23 +426,51 @@ def get_initial_speed(self, veh_id):
     def get_observed_pedestrians(self, veh_id, pedestrians, radius=50):
         position = self.get_orientation(veh_id)[:2]
         orientation = self.get_orientation(veh_id)[2]
-        orientation += 90 # orientation is 0 when facing North in SUMO, but 0 when facing EAST in util.py
         observed_pedestrians = []
         for ped_id in pedestrians.get_ids():
             if util.observed(position, orientation, pedestrians.get_position(ped_id), looking_distance=radius):
                 observed_pedestrians.append(ped_id)
         return observed_pedestrians
 
-    def get_observed_vehicles(self, veh_id, vehicles, radius=50):
+    def get_observed_vehicles(self, veh_id, radius=50):
+        position = self.get_orientation(veh_id)[:2]
+        orientation = self.get_orientation(veh_id)[2]
+        observed_vehicles = []
+        for v_id in self.get_ids():
+            if util.observed(position, orientation, self.get_orientation(v_id)[:2], looking_distance=radius):
+                observed_vehicles.append(v_id)
+        return observed_vehicles
+
+    def get_viewable_objects(self, veh_id, pedestrians=None, radius=50):
+        viewable_pedestrians, viewable_vehicles = [], []
+        observed_vehicles = []
+        blocked = []
+
         position = self.get_orientation(veh_id)[:2]
         orientation = self.get_orientation(veh_id)[2]
-        orientation += 90 # orientation is 0 when facing North in SUMO, but 0 when facing EAST in util.py
-        obs_vehs = []
-        for veh_id in vehicles.get_ids():
-            print(vehicles.get_position(veh_id))
-            if util.observed(position, orientation, vehicles.get_orientation(veh_id)[:2], looking_distance=radius):
-                obs_vehs.append(veh_id)
-        return obs_vehs
+
+        for v_id in self.get_ids():
+            if util.observed(position, orientation, self.get_orientation(v_id)[:2], looking_distance=radius) and v_id != veh_id:
+
+                observed_vehicles.append(v_id)
+                blocked.append(util.get_blocked_segments(position,
+                    self.get_orientation(v_id)[:2],
+                    self.get_orientation(v_id)[2],
+                    self.get_length(v_id),
+                    self.get_width(v_id)))
+
+        #TODO: vehicle often blocks itself out of view
```

Comment:
Reminder to complete this TODO. Also tag TODOs with your initials so we know who to hunt down